### PR TITLE
Fix the lowercase of compute resources

### LINKED ISSUE #792: [DOC] the category's value is not right
Hi there,

Thank you for opening an issue. Please note that we try to keep the Databricks Provider issue tracker reserved for bug reports and feature requests. For general Terraform questions, please see: https://www.terraform.io/community.html.
This template is for both adding enhancement as well as pointing out issues with the provider documentation.

### Affected Resource(s)
https://github.com/databrickslabs/terraform-provider-databricks/blob/6b702f5803003cf1d10f6f11dc39e63426184f47/docs/data-sources/node_type.md

in this part, the category is not right
```
category - (Optional) Node category, which can be one of:
General purpose
Memory optimized
Storage optimized
Compute optimized
```
`
the second word should be capitalized. From my understanding, it should match the category  property returned by the api 
/api/2.0/clusters/list-node-types
restAPI_result
,pls check the snapshot in the attachment 
`
### List of things to potentially add/remove:
This is a list of things to manipulate in the docs:
- [ ] General purpose  => General Purpose
- [ ] Memory optimized => Memory Optimized 
- [ ] Storage optimized  => Storage Optimized 
- [ ] Compute optimized => Compute Optimized

### PR DESCRIPTION
**Pull request by KoenR3**
Camelcase the category of a resource as this is what the platform expects.

This fixes  #792

### CODE REVIEW

**Comment by nfx** [comment]
@KoenR3 did you test it on all of AWS, Azure, and GCP workspaces?..

@alexott , could you review this, as you made the most recent changes there

**Comment by alexott** [comment]
Current list on Azure is:

```
>curl -s -n -X GET "$DATABRICKS_HOST/api/2.0/clusters/list-node-types" -H "Authorization: Bearer $DATABRICKS_TOKEN"|jq '.node_types[]|.category'|sort |uniq -c
  14 "Compute Optimized"
   9 "GPU Accelerated"
   7 "General Purpose (HDD)"
  33 "General Purpose"
   4 "Memory Optimized (Remote HDD)"
  51 "Memory Optimized"
   9 "Storage Optimized"
```

I think that it makes sense  to use case-insensitive comparison in the code. I'll fix that

File: docs/data-sources/node_type.md [hunk 1]
```diff
@@ -46,10 +46,10 @@ Data source allows you to pick groups by the following attributes
...
 * `category` - (Optional) Node category, which can be one of:
-  * `General purpose`
-  * `Memory optimized`
-  * `Storage optimized`
-  * `Compute optimized`
+  * `General Purpose`
+  * `Memory Optimized`
+  * `Storage Optimized`
+  * `Compute Optimized`
   * `GPU`
```
Comment (alexott) [inline docs/data-sources/node_type.md:53]: it should be `GPU Accelerated`.

**Review by alexott — CHANGES_REQUESTED** [review]
see comment on GPU

It would be also useful to incorporate differences between clouds & show the command that could be used to extract the list of possible values, as this list changes from time to time (like, if we start support GPUs on GCP, etc.)

File: docs/data-sources/node_type.md (hunk 1 above)
Comment (nfx) [reply docs/data-sources/node_type.md:53]: @KoenR3 ^^

**Comment by alexott** [comment]
@nfx can you merge this PR, I'll add clarifications on top of it.

**Comment by KoenR3** [comment]
Sorry about the late reply, swamped with work...

**Comment by alexott** [comment]
no problem, understood...

**Pull request merged**